src/components/layout/header/header.tsx

```diff
@@ -161,6 +162,7 @@ const Header: React.FC<HeaderProps> = (props) => {
     <StyledHeader
       $lightTheme={Boolean(props.$lightTheme)}
       $transparent={isHeaderTransparent}
+      onKeyDown={(e) => e.keyCode === 27 && setIsNavigationVisible(false)}
```

Comment:
I tried to place the keyDown listener inside the `NavigationFlyout` component, but then the `CloseBurgerMenuIcon` focuses first on ESC keypress (even with `tabIndex`)

Reply:
This works, thank you!

You  can also attach and detach a global event listener for `keyup' or `keydown` on `document`. That way it only listens when it's open. I fear some side effects here. If you feel confident let's merge.

Reply:
did you try a global approach or checked side effects of your current solution? Will it hijack the ESC key when in a form (in the content) or such ? The PR is fine as is now but I would like to hear some feedback from you.

Reply:
I compared the forms with those from the current version and the ESC key has no side effects. I have not tried a global approach, do you think this would be better?